sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -882,6 +882,45 @@ def my_pipeline() -> NamedTuple('Outputs', [
             ]):
                 task = print_and_return(text='Hello')
 
+    def test_pipeline_reusing_other_pipeline_multiple_times(self):
+
+        @dsl.pipeline()
+        def reusable_pipeline():
+            print_op(message='Reused pipeline')
+
+        @dsl.pipeline()
+        def do_something_else_pipeline():
+            reusable_pipeline()
+
+        @dsl.pipeline()
+        def orchestrator_pipeline():
+            reusable_pipeline()
+
+            do_something_else_pipeline()
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            output_yaml = os.path.join(tmpdir, 'result.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=orchestrator_pipeline, package_path=output_yaml)
+            self.assertTrue(os.path.exists(output_yaml))
+
+            with open(output_yaml, 'r') as f:
+                pipeline_spec = yaml.safe_load(f)
+                tasks = [
+                    comp.get('dag', {}).get('tasks', {})
+                    for comp in pipeline_spec['components'].values()
+                ]
+                component_refs = [[
+                    x.get('componentRef', {}).get('name')
+                    for x in task.values()
+                ]
+                                  for task in tasks]
+                all_component_refs = [
+                    item for sublist in component_refs for item in sublist
+                ]
+                counted_refs = collections.Counter(all_component_refs)
+                self.assertEqual(1, max(counted_refs.values()))
```

Comment:
In my local testing, this passes both before and after the changes. I also observe that the YAML is not different. Is it possible there is something missing from this test case?

Reply:
My bad - went to harsh on cutting components to a minimal version.